### PR TITLE
common script for contract deployment

### PR DESCRIPTION
**Pull request by shivamgarglab49**
common script for contract deployment

### CODE REVIEW

File: deployment/Helper.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { getAllFilesSync } from "get-all-files";
+
+export class Helper {
+  static getFileNameFromPath(path: string): string {
```
Comment (amandeeprehal-lab49) [inline deployment/Helper.ts:4]: parseFileName or extractFileName looks better name for me.

File: deployment/contract/deploy.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  "GovernorTextProposal",
+  "GovernorTokenCreate",
+  "TransparentUpgradeableProxy",
+];
+
+async function main() {
+  const contractName = (
+    await inquirer.prompt([
+      {
+        type: "rawlist",
```
Comment (amandeeprehal-lab49) [inline deployment/contract/deploy.ts:22]: Duplicate code - line 20 and line 34
```await inquirer.prompt([
      {
        type: "rawlist",
        name: "contractName",
        message: "Please select which contract you want to deploy ? ",
        choices: [...SUPPORTED_CONTRACTS_FOR_DEPLOYMENT, "exit"],
      },
    ])```

Comment (amandeeprehal-lab49) [reply deployment/contract/deploy.ts:22]: using function parameter and create a new function.

File: deployment/contract/deploy.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        name: "option",
+        message: "Please select any option for proxy operation from menu ... ",
+        choices: ["create", "update", "exit"],
+      },
+    ])
+  ).option;
+  if (option === "create") {
```
Comment (amandeeprehal-lab49) [inline deployment/contract/deploy.ts:43]: How about Switch statement here?

File: deployment/logicCode.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import dotenv from "dotenv";
+dotenv.config();
+
+const deployment = new Deployment();
+
+export async function main(_contractName: string | null = null) {
+  const contractName = (
+    _contractName ?? process.env.CONTRACT_NAME!
+  ).toLowerCase();
+  const compiledPaths = Helper.getContractPathList("./artifacts").compiledPaths;
```
Comment (amandeeprehal-lab49) [inline deployment/logicCode.ts:13]: Move this to separate function -
```
const compiledPaths = Helper.getContractPathList("./artifacts").compiledPaths;
  const filePath = compiledPaths.find(
    (path) => Helper.getFileNameFromPath(path).toLowerCase() === contractName
  );
  if (filePath === undefined) {
    return `Failed to locate (${contractName}) contract json`;
  }
```

File: deployment/logicCode.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import { Helper } from "./Helper";
+import { Deployment } from "./deployContractOnTestnet";
+import { ContractFunctionParameters } from "@hashgraph/sdk";
+import dotenv from "dotenv";
+dotenv.config();
+
+const deployment = new Deployment();
+
+export async function main(_contractName: string | null = null) {
```
Comment (amandeeprehal-lab49) [inline deployment/logicCode.ts:9]: You can also write _contractName?: string using optional

File: deployment/transparentUpgradeableProxyCode.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  ContractFunctionParameters
+} from "@hashgraph/sdk";
+import { Deployment } from "./deployContractOnTestnet";
+import { DeployedContract } from "./model/contract";
+import { ContractService } from "./service/ContractService";
+dotenv.config();
+
+const contractService = new ContractService();
+
+export async function main(_contractName: string | null = null) {
```
Comment (amandeeprehal-lab49) [inline deployment/transparentUpgradeableProxyCode.ts:13]: Same here optional use

File: deployment/upgradeProxyCode.ts [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+            .addAddress(newImplementation)
+        )
+        .freezeWith(client)
+        .sign(adminKey);
+      const liquidityPoolTx = await liquidityPool.execute(client);
+      const transferTokenRx = await liquidityPoolTx.getReceipt(client);
+      console.log(`upgradedTo: ${transferTokenRx.status}`);
+  };
+  
+  export async function main(_contractName: string | null = null) {
```
Comment (amandeeprehal-lab49) [inline deployment/upgradeProxyCode.ts:82]: Same optional

File: entrypoint.sh [hunk 8]
```diff
@@ -47,7 +47,7 @@ echo CONTRACT_NAME=$CONTRACT_NAME >> .env;
 if [ "$CONTRACT_TYPE" = "Logic" ]; then 
     echo "Running contract deployment ........ " $CONTRACT_NAME;
     contract=$CONTRACT_NAME.ts;
-    npx hardhat run ./deployment/contract/$contract;
+    npx hardhat run ./deployment/logic.ts;
```
Comment (amandeeprehal-lab49) [inline entrypoint.sh:50]: Have you tested this change? We need to test from GitHub.

File: deployment/Helper.ts [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    return path.split("/").at(-1)?.split(".")[0] || "";
+  }
+
+  static getContractPathList(path:string) {
```
[the remark this answers is not in the document]
Comment (shivamgarglab49) [reply deployment/Helper.ts:12]: done

File: deployment/contract/deploy.ts (hunk 3 above)
Comment (shivamgarglab49) [reply deployment/contract/deploy.ts]: if else statement is more clear here.

File: entrypoint.sh (hunk 8 above)
Comment (shivamgarglab49) [reply entrypoint.sh:50]: yes done.

**Review by amandeeprehal-lab49 — APPROVED** [review]

**Pull request merged**